### PR TITLE
[plugin.video.filmarkivet] 3.1.0

### PR DESCRIPTION
**Pull request by pugo**
### Description
Updated to support https URI:s now used by www.filmarkivet.se. Changed from urllib to requests for better Python 2/3 support.
### Checklist:
- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.
- If you see no activity on your PR after a week (so at least one weekend has passed) then please go to the #kodi-dev freenode IRC channel to reach out to the team

### CODE REVIEW

File: plugin.video.filmarkivet/addon.xml [hunk 1]
```diff
@@ -1,11 +1,13 @@
 <?xml version="1.0" encoding="UTF-8" standalone="yes"?>
 <addon id="plugin.video.filmarkivet"
        name="Filmarkivet"
-       version="3.0.0"
+       version="3.1.0"
        provider-name="Anders Piniesjö (pugo@pugo.org)">
   <requires>
-    <import addon="xbmc.python" version="2.14.0"/>
+    <import addon="xbmc.python" version="2.25.0"/>
```
Comment (enen92) [inline plugin.video.filmarkivet/addon.xml]: Gotham needs 2.14.0

File: plugin.video.filmarkivet/addon.xml [hunk 2]
```diff
@@ -20,10 +22,19 @@
...
+    <summary lang="de">Filme vom Filmarchiv des schwedisches Filminstituts</summary>
     <description lang="en">Filmarkivet is an archive of historical Swedish films, provided by the Swedish Film Institute.</description>
     <description lang="sv">Filmarkivet är ett arkiv med historiska svenska filmer, publicerade av Svenska Filminstitutet.</description>
     <description lang="de">Filmarkivet ist ein Archiv von historischen schwedischen Filmen, veröffentlicht bei Svenska Filminstitutet.</description>
-    <news>v3.0.0 (2016-10-24)[CR]- Total rewrite to match new Filmarkivet site.[CR]- Caching gives huge speed-up.</news>
+    <news>v3.1.0 (2019-12-29)
+      [fix] Support Filmarkivet web changes
+    </news>
+    <assets>
+      <icon>resources/icon.png</icon>
```
Comment (enen92) [inline plugin.video.filmarkivet/addon.xml]: Please keep the icon in root directory for backward compatibility with gotham

File: plugin.video.filmarkivet/addon.xml [hunk 3]
```diff
@@ -20,10 +22,19 @@
     <website></website>
```
Comment (enen92) [inline plugin.video.filmarkivet/addon.xml]: Drop website tag if empty

File: plugin.video.filmarkivet/default.py [hunk 4]
```diff
@@ -18,104 +18,103 @@
...
-		li = xbmcgui.ListItem(title, iconImage=item.icon)
-		li.setInfo('video', {'title': item.title, 'year': item.year, 'duration': item.duration})
-		if item.playable:
-			li.setProperty('isplayable', 'true')
-			li.setInfo('video', {'plot': item.description})
-		xbmcplugin.addDirectoryItem(info.handle, item.url, li, not item.playable)
-	xbmcplugin.endOfDirectory(info.handle)
+    for item in menu:
+        title = u'{} | {}'.format(item.title, item.description) if item.description else item.title
+        li = xbmcgui.ListItem(title, iconImage=item.icon)
```
Comment (enen92) [inline plugin.video.filmarkivet/default.py]: Drop iconImage, use setArt only

File: plugin.video.filmarkivet/addon.xml [hunk 5]
```diff
@@ -20,10 +22,19 @@
...
     <description lang="en">Filmarkivet is an archive of historical Swedish films, provided by the Swedish Film Institute.</description>
     <description lang="sv">Filmarkivet är ett arkiv med historiska svenska filmer, publicerade av Svenska Filminstitutet.</description>
     <description lang="de">Filmarkivet ist ein Archiv von historischen schwedischen Filmen, veröffentlicht bei Svenska Filminstitutet.</description>
-    <news>v3.0.0 (2016-10-24)[CR]- Total rewrite to match new Filmarkivet site.[CR]- Caching gives huge speed-up.</news>
+    <news>v3.1.0 (2019-12-29)
+      [fix] Support Filmarkivet web changes
+    </news>
+    <assets>
+      <icon>resources/icon.png</icon>
+      <fanart></fanart>
```
Comment (enen92) [inline plugin.video.filmarkivet/addon.xml]: Please remove all if empty

**Comment by pugo** [comment]
Updated according to comments. Thanks for good review!

**Comment by enen92** [comment]
Please check travis failure reasons:

1) Strings.xml is no longer supported. you need to upgrade to gettext. There is an helper addon that will do most of the job for you: https://forum.kodi.tv/showthread.php?tid=332789

2) The addon is in gotham and the minimum requests version there doesn't match: script.module.requests, required: 2.22.0, Available: 2.15.1

3) same goes for requests cache. Unfortunately in this case the addon is not available for gotham: script.module.requests-cache is not available in current repository

I'd say you'd be better moving the addon to an upper repository (let's say krypton) and mark the version in gotham as broken

File: plugin.video.filmarkivet/resources/language/resource.language.en_gb/strings.po [hunk 6]
```diff
@@ -0,0 +1,54 @@
+# Kodi Media Center language file
```
Comment (enen92) [inline plugin.video.filmarkivet/resources/language/resource.language.en_gb/strings.po]: Sorry for so many requests but, for the addon to be in the gotham branch you need to keep the old language layout while moving to gettext:

plugin.video.filmarkivet/resources/language/English/strings.po

File: plugin.video.filmarkivet/resources/language/resource.language.sv_se/strings.po [hunk 7]
```diff
@@ -0,0 +1,54 @@
+# Kodi Media Center language file
```
Comment (enen92) [inline plugin.video.filmarkivet/resources/language/resource.language.sv_se/strings.po]: Sorry for so many requests but, for the addon to be in the gotham branch you need to keep the old language layout while moving to gettext:

plugin.video.filmarkivet/resources/language/Swedish/strings.po

File: plugin.video.filmarkivet/resources/language/resource.language.en_gb/strings.po (hunk 6 above)
Comment (pugo) [reply plugin.video.filmarkivet/resources/language/resource.language.en_gb/strings.po]: No problem, it is my fault as I haven't updated the addon very often.
Ok, I will update it to Krypton then, I agree that it should be easier.
If I move to Krypton, then I don't need to keep the old language layout?
I guess I should cancel this PR then and make a new for Krypton?

**Comment by pugo** [comment]
Closing this pull request, as I'm moving to Krypton for lowest supported version.

**Pull request closed without merging**